airflow/providers/databricks/sensors/databricks_partition.py

```diff
@@ -0,0 +1,157 @@
+        table_name: str = "",
+        partition_name: dict,
+        handler: Callable[[Any], Any] = fetch_one_handler,
+        caller: str = "DatabricksPartitionSensor",
+        client_parameters: dict[str, Any] | None = None,
+        partition_operator: str = "=",
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.databricks_conn_id = databricks_conn_id
+        self._http_path = http_path
+        self._sql_endpoint_name = sql_endpoint_name
+        self.session_config = session_configuration
+        self.http_headers = http_headers
+        self.catalog = catalog
+        self.schema = schema
+        self.table_name = table_name
+        self.partition_name = partition_name
+        self.caller = caller
+        self.client_parameters = client_parameters or {}
+        self.hook_params = kwargs.pop("hook_params", {})
+        self.handler = handler
+        self.partition_operator = partition_operator
+
+    def _get_hook(self) -> DatabricksSqlHook:
+        return DatabricksSqlHook(
+            self.databricks_conn_id,
+            self._http_path,
+            self._sql_endpoint_name,
+            self.session_config,
+            self.http_headers,
+            self.catalog,
+            self.schema,
+            self.caller,
+            **self.client_parameters,
+            **self.hook_params,
+        )
+
+    def _sql_sensor(self, sql):
+        hook = self._get_hook()
+        sql_result = hook.run(
+            sql,
+            handler=self.handler if self.do_xcom_push else None,
+        )
+        return sql_result
+
+    def _check_table_partitions(self) -> list:
+        if self.catalog is not None:
+            complete_table_name = str(self.catalog + "." + self.schema + "." + self.table_name)
+            self.log.debug("Table name generated from arguments: %s", complete_table_name)
+        else:
+            raise AirflowException("Catalog name not specified, aborting query execution.")
+        partition_columns = self._sql_sensor(f"describe detail {complete_table_name}")[7]
+        self.log.debug("table_info: %s", partition_columns)
+        if len(partition_columns) < 1:
+            raise AirflowException("Table %s does not have partitions", complete_table_name)
+        partitions_list = []
+        for partition_col, partition_value in self.partition_name.items():
+            if partition_col in partition_columns:
+                if isinstance(partition_value, list):
+                    partitions_list.append(f"""{partition_col} in {tuple(partition_value)}""")
+                if isinstance(partition_value, (int, float, complex)):
+                    partitions_list.append(f"""{partition_col}{self.partition_operator}{partition_value}""")
+                if isinstance(partition_value, str):
+                    partitions_list.append(
+                        f"""{partition_col}{self.partition_operator}\"{partition_value}\""""
+                    )
```

Comment:
it's better to use formatting functions from sql connector, similar to what is done in the `DatabricksCopyIntoOperator` - otherwise we have a chance of getting SQL injection.

Reply:
@alexott For this change, I am thinking of generating the SQL instead of receiving it as an argument, to prevent injections: `SELECT {expression_list, formatted — list} FROM {table_name} (generated by arguments) WHERE {filter_vals, formatted — dict} GROUP BY {group_by vals, formatted — list} ORDER BY {order_by vals, formatted — list}` where the where , group by and order by are optional, but will be validated by the formatter if it receives values for the arguments in the class.

Thought of two options: Generating the SQL like the above, which has a downside of restricting the SQLs we can allow to be passed to the sensor, but it also prevents unexpected injections as a result.

Another option is to breakdown the SQL received from the user and do the validation via the SQL formatter before executing it via the hook.

I am leaning towards the first option, any thoughts/preferences or alternative suggestions?